datalake/src/main/java/com/sequenceiq/datalake/flow/freeipa/upscale/handler/FreeIpaUpscaleWaitHandler.java

```diff
@@ -0,0 +1,113 @@
+    private int sleepTimeInSec;
+
+    @Value("${sdx.freeipa.upscale.duration_min:60}")
+    private int durationInMinutes;
+
+    @Inject
+    private FreeipaService freeipaService;
+
+    @Override
+    public String selector() {
+        return "FreeIpaUpscaleWaitRequest";
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<FreeIpaUpscaleWaitRequest> event) {
+        return new FreeIpaUpscaleFailedEvent(resourceId, event.getData().getUserId(), e);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<FreeIpaUpscaleWaitRequest> event) {
+
+        FreeIpaUpscaleWaitRequest freeIpaUpscaleWaitRequest = event.getData();
+        Long sdxId = freeIpaUpscaleWaitRequest.getResourceId();
+        String userId = freeIpaUpscaleWaitRequest.getUserId();
+        Selectable response;
+        try {
+            LOGGER.debug("Start polling freeIpa upscale process for id: {}", sdxId);
+            PollingConfig pollingConfig = new PollingConfig(sleepTimeInSec, TimeUnit.SECONDS, durationInMinutes, TimeUnit.MINUTES);
+            waitFreeIpaUpscale(freeIpaUpscaleWaitRequest.getEnvCrn(), freeIpaUpscaleWaitRequest.getOperationId(), pollingConfig);
+            response = new FreeIpaUpscaleSuccessEvent(sdxId, userId);
+        } catch (UserBreakException userBreakException) {
+            LOGGER.error("Upscale polling exited before timeout. Cause: ", userBreakException);
+            response = new FreeIpaUpscaleFailedEvent(sdxId, userId, userBreakException);
+        } catch (PollerStoppedException pollerStoppedException) {
+            LOGGER.error("Upscale poller stopped for stack: {}", sdxId);
+            response = new FreeIpaUpscaleFailedEvent(sdxId, userId,
+                    new PollerStoppedException("FreeIpa Upgrade timed out after " + durationInMinutes + "minues"));
+        } catch (PollerException exception) {
+            LOGGER.error("Upscale polling failed for stack: {}", sdxId);
+            response = new FreeIpaUpscaleFailedEvent(sdxId, userId, exception);
+        }
+        return response;
+    }
+
+    public void waitFreeIpaUpscale(String envCrn, String operationId, PollingConfig pollingConfig) {
+        String accountId = Crn.safeFromString(envCrn).getAccountId();
+        Polling.waitPeriodly(pollingConfig.getSleepTime(), pollingConfig.getSleepTimeUnit())
+                .stopIfException(pollingConfig.getStopPollingIfExceptionOccurred())
+                .stopAfterDelay(pollingConfig.getDuration(), pollingConfig.getDurationTimeUnit())
+                .run(() -> {
+                    LOGGER.info("Deletion polling freeIpa for operation status: '{}' in '{}' env", operationId, envCrn);
+                    Optional<OperationStatus> freeIpaOperation = ThreadBasedUserCrnProvider.doAsInternalActor(() ->
+                            freeipaService.getFreeIpaOperation(operationId, accountId));
+                    if (freeIpaOperation.isEmpty()) {
```

Comment:
if this remains true, this would poll for 60 min, e.g. 5 retry would be better before failing

Reply:
there case I was thinking we might get a not found is if we had just called upscale before the operation had been registered. I could see there might be an issue with the initial call, but we wouldn't have the operationId to poll on in that case. Also how long is an upscale expected to take?